decisions/ui-test-accessibility-identifiers.md

```diff
@@ -0,0 +1,39 @@
+# UI test accessibility identifiers
+
+## Decision
+* **Standalone Components**: No accessibility identifiers for UI tests. Consumers can add them as needed.
+* **Components in Larger UIs**:
+  * **Hardcoded identifiers** if the component is used in a single place.
+  * **Dynamic identifiers** if the component is reused in multiple places.
+
+## Thinking
+
+### Standalone components
+Standalone components are components that are not part of a larger UI. They are typically used in isolation and are not composed of other components. Examples of standalone components are `BpkButton`, `BpkText`, `BpkIcon`, etc.
+
+These components can receive accessibility identifiers from the consumer if they need to be tested in a UI test. This is because the consumer is the one that knows the context in which the component is being used.
+
+**Example**
+
+```swift
+BPKButton("Search") {
+    /* Tap code */
+}
+.accessibilityIdentifier("search_button")
+```
+
+### Components that are part of a larger UI
+Components that are part of a larger UI are typically composed of other components. Examples of these components are `BpkAppSearchModal`, `BpkImageGalleryImageGrid`, `BpkNudger`, etc.
+
+These components contain nested interactive elements and need to be exposed to UI tests. We need to add accessibility identifiers to these components to make them testable.
+Based on how we expect the component to be used we can decide whether to add hardcoded or dynamic accessibility identifiers.
```

Comment:
@gert-janvercauteren 
just one question
do you think we need expose an accessibility identifiers to UI tests for "container" components like BpkAppSearchModal? As they're not taking any functional purpose, they are not controls or having any values. So my thought is to recommend to not add any accessibility identifiers there. Or are there any cases where it could be useful?

Reply:
Interesting point! The idea for 'container components' is to expose the interactive controls within. I'll update the doc to reflect that 🙇